File: 240306/중앙값 계산 2/get-median-2.py
n = int(input())
a = [int(x) for x in input().split()]
b = []
ans = []
for i in range(n+1):
  if i % 2 == 1:
    b.append(a[:i])
    b.sort()

for j in range(len(b)):
  b[j].sort()
  ans.append(b[j][j])

for _ in ans:
  print(_, end = ' ')

File: 240421/중앙값 계산 2/get-median-2.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)
const N = Number(input[0])
const numbers = input[1].split(' ').map(Number)

let numList = []
let answer = ''
for (let i = 0; i < N; i++) {
    numList.push(numbers[i])
    if (i % 2 === 0) {
        numList.sort((a,b) => a-b)
        answer += (numList[parseInt((numList.length)/2)]) + ' '
    }
}
console.log(answer)